### PR TITLE
For #25025 - Integrate Pocket sponsored stories

### PR DESCRIPTION
**Pull request by Mugurell**
~Needs https://github.com/mozilla-mobile/android-components/pull/12070~
~Needs https://github.com/mozilla-mobile/android-components/pull/12183~

~Based on https://github.com/mozilla-mobile/fenix/pull/24892.~

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by Mugurell** [comment]
Rebased to fix conflicts.

**Comment by Mugurell** [comment]
In the meantime the design changed a little. Figma link available on Jira.
Updated the code now to comply.

File: app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesComposables.kt [hunk 1]
```diff
@@ -114,21 +118,78 @@ fun PocketStory(
...
+            maxLines = 2,
+        )
+
+        Spacer(Modifier.height(9.dp))
+
+        Row(verticalAlignment = Alignment.CenterVertically) {
+            Icon(
+                painter = painterResource(id = R.drawable.pocket_star_stroke),
+                contentDescription = null,
+                tint = FirefoxTheme.colors.iconSecondary,
```
Comment (Mugurell) [inline app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesComposables.kt:153]: This and the text color below are slightly different than on Figma to not have to add a custom color.
Asked Sam from UX for validating any one approach.

File: app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt [hunk 2]
```diff
@@ -54,10 +55,26 @@ sealed class AppAction : Action {
...
+    /**
+     * Indicates the given [storiesShown] were seen by the user.
+     */
+    data class PocketStoriesShown(val storiesShown: List<PocketStory>) : AppAction()
+    /**
+     * Clean all in-memory data about Pocket stories and categories.
+     */
+    object PocketStoriesClean : AppAction()
+    /**
+     * Replace the current list of Pocket sponsored stories.
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt:75]: nit: `Replaces` to follow the pattern in other actions doc.

File: app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt [hunk 3]
```diff
@@ -54,10 +55,26 @@ sealed class AppAction : Action {
...
+     */
     data class DeselectPocketStoriesCategory(val categoryName: String) : AppAction()
-    data class PocketStoriesShown(val storiesShown: List<PocketRecommendedStory>) : AppAction()
-    data class PocketStoriesChange(val pocketStories: List<PocketRecommendedStory>) : AppAction()
+    /**
+     * Indicates the given [storiesShown] were seen by the user.
+     */
+    data class PocketStoriesShown(val storiesShown: List<PocketStory>) : AppAction()
+    /**
+     * Clean all in-memory data about Pocket stories and categories.
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt:71]: nit: `Cleans` to follow the pattern in other actions doc.

File: app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesComposables.kt (hunk 1 above)
Comment (Mugurell) [reply app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesComposables.kt:153]: Apparently the slightly different color was just a design quirk.
Okayed to use the same `textSecondary` / `iconSecondary` colors.

File: app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesViewHolder.kt [hunk 4]
```diff
@@ -23,7 +23,7 @@ import androidx.compose.ui.unit.dp
 import androidx.lifecycle.LifecycleOwner
 import androidx.recyclerview.widget.RecyclerView
 import mozilla.components.lib.state.ext.observeAsComposableState
-import mozilla.components.service.pocket.PocketRecommendedStory
+import mozilla.components.service.pocket.PocketStory.PocketRecommendedStory
 import org.mozilla.fenix.R
 import org.mozilla.fenix.R.string
```
Comment (mcarare) [inline app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesViewHolder.kt:28]: Not from this change, but we probably do not want this import.

**Comment by Mugurell** [comment]
Thank you Mihai!
Addressed now your comments and rebased. Should be good to launch tomorrow along with https://github.com/mozilla-mobile/android-components/pull/12183.

**Comment by Mugurell** [comment]
Restarting CI.

**Comment by Mugurell** [comment]
`toolchain-linux64-android-gradle-dependencies ` failed again because of:
```
Execution failed for task ':app:packageDebug'.
> Multiple task action failures occurred:
   > A failure occurred while executing com.android.build.gradle.tasks.PackageAndroidArtifact$IncrementalSplitterRunnable
      > Self-suppression not permitted
   > A failure occurred while executing com.android.build.gradle.tasks.PackageAndroidArtifact$IncrementalSplitterRunnable
      > Java heap space
   > A failure occurred while executing com.android.build.gradle.tasks.PackageAndroidArtifact$IncrementalSplitterRunnable
      > Self-suppression not permitted
   > A failure occurred while executing com.android.build.gradle.tasks.PackageAndroidArtifact$IncrementalSplitterRunnable
      > Java heap space
```

Restarting CI.

**Pull request merged**